Exception in completion when there is an unimported enum

eg. Below, `GenerationType` is not imported:

```java
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class FollowGraph {

	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "id")
	private User follower;

	@ManyToOne
	@JoinColumn(name = "id")
	private User followed;

	public FollowGraph() {}

	public User getFollower() {
		return this.follower;
	}

	public void setFollower(User follower) {
		this.follower = follower;
	}

	public User getFollowed() {
		return this.followed;
	}

	public void setFollowed(User followed) {
		this.followed = followed;
	}

}
```

log from jdt-ls:

```
Error - 11:22:05 AM] Request textDocument/codeAction failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.dom.IBinding.getName()" because "<parameter1>" is null
	at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:323)
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:359)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:364)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:693)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:527)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.dom.IBinding.getName()" because "<parameter1>" is null
	at org.eclipse.jdt.internal.codeassist.DOMCompletionEngine.lambda$28(DOMCompletionEngine.java:428)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:196)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$7$1FlatMap.test(ReferencePipeline.java:307)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.HashMap$KeySpliterator.tryAdvance(HashMap.java:1737)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.allMatch(ReferencePipeline.java:677)
	at java.base/java.util.stream.ReferencePipeline$7$1FlatMap.accept(ReferencePipeline.java:292)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1034)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:687)
	at org.eclipse.jdt.internal.codeassist.DOMCompletionEngine.run(DOMCompletionEngine.java:429)
	at org.eclipse.jdt.internal.core.CompilationUnit.codeComplete(CompilationUnit.java:461)
	at org.eclipse.jdt.internal.core.CompilationUnit.codeComplete(CompilationUnit.java:450)
	at org.eclipse.jdt.internal.ui.text.correction.SimilarElementsRequestor.process(SimilarElementsRequestor.java:168)
	at org.eclipse.jdt.internal.ui.text.correction.SimilarElementsRequestor.findSimilarElement(SimilarElementsRequestor.java:113)
	at org.eclipse.jdt.internal.ui.text.correction.UnresolvedElementsBaseSubProcessor.addSimilarTypeProposals(UnresolvedElementsBaseSubProcessor.java:918)
	at org.eclipse.jdt.internal.ui.text.correction.UnresolvedElementsBaseSubProcessor.collectVariableProposals(UnresolvedElementsBaseSubProcessor.java:338)
	at org.eclipse.jdt.ls.core.internal.corrections.proposals.UnresolvedElementsSubProcessor.getVariableProposals(UnresolvedElementsSubProcessor.java:62)
	at org.eclipse.jdt.ls.core.internal.corrections.QuickFixProcessor.process(QuickFixProcessor.java:162)
	at org.eclipse.jdt.ls.core.internal.corrections.QuickFixProcessor.getCorrections(QuickFixProcessor.java:92)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionCommands(CodeActionHandler.java:190)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$14(JDTLanguageServer.java:759)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:690)
	... 5 more
```

seems related to bindings

--- a/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/DOMCompletionEngine.java
+++ b/org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/DOMCompletionEngine.java
@@ -430,27 +430,29 @@ public class DOMCompletionEngine implements Runnable {
 					suggestDefaultCompletions = false;
 				}
 				if (context.getLocationInParent().getId().equals(QualifiedName.QUALIFIER_PROPERTY.getId()) && context.getParent() instanceof QualifiedName) {
-					// eg.
-					// void myMethod() {
-					//   String myVariable = "hello, mom";
-					//   myVariable.|
-					//   Object myObj = null;
-					// }
-					// It thinks that our variable is a package or some other type. We know that it's a variable.
-					// Search the scope for the right binding
 					IBinding incorrectBinding = ((SimpleName) context).resolveBinding();
-					Bindings localBindings = new Bindings();
-					scrapeAccessibleBindings(localBindings);
-					Optional<IVariableBinding> realBinding = localBindings.stream() //
-							.filter(IVariableBinding.class::isInstance)
-							.map(IVariableBinding.class::cast)
-							.filter(varBind -> varBind.getName().equals(incorrectBinding.getName()))
-							.findFirst();
-					if (realBinding.isPresent()) {
-						processMembers(context, realBinding.get().getType(), specificCompletionBindings, false);
-						this.prefix = ""; //$NON-NLS-1$
-						publishFromScope(specificCompletionBindings);
-						suggestDefaultCompletions = false;
+					if (incorrectBinding != null) {
+						// eg.
+						// void myMethod() {
+						//   String myVariable = "hello, mom";
+						//   myVariable.|
+						//   Object myObj = null;
+						// }
+						// It thinks that our variable is a package or some other type. We know that it's a variable.
+						// Search the scope for the right binding
+						Bindings localBindings = new Bindings();
+						scrapeAccessibleBindings(localBindings);
+						Optional<IVariableBinding> realBinding = localBindings.stream() //
+								.filter(IVariableBinding.class::isInstance)
+								.map(IVariableBinding.class::cast)
+								.filter(varBind -> varBind.getName().equals(incorrectBinding.getName()))
+								.findFirst();
+						if (realBinding.isPresent()) {
+							processMembers(context, realBinding.get().getType(), specificCompletionBindings, false);
+							this.prefix = ""; //$NON-NLS-1$
+							publishFromScope(specificCompletionBindings);
+							suggestDefaultCompletions = false;
+						}
 					}
 				}
 			}
@@ -694,13 +696,15 @@ public class DOMCompletionEngine implements Runnable {
 							? IJavaSearchConstants.ANNOTATION_TYPE
 							: IJavaSearchConstants.TYPE;
 					ExtendsOrImplementsInfo extendsOrImplementsInfo = isInExtendsOrImplements(this.toComplete);
-					findTypes(completeAfter, typeMatchRule, null)
+					if (!this.requestor.isIgnored(CompletionProposal.TYPE_REF)) {
+						findTypes(completeAfter, typeMatchRule, null)
 							.filter(type -> this.pattern.matchesName(this.prefix.toCharArray(),
 									type.getElementName().toCharArray()))
 							.filter(type -> {
 								return filterBasedOnExtendsOrImplementsInfo(type, extendsOrImplementsInfo);
 							})
 							.map(this::toProposal).forEach(this.requestor::accept);
+					}
 				}
 				checkCancelled();
 				suggestPackages();

